test/test.js

```diff
@@ -1,151 +1,210 @@
+  request = require('superagent')
+  request.http2 = true
+}
 
 describe('typeis(req, type)', function () {
-  it('should ignore params', function () {
-    var req = createRequest('text/html; charset=utf-8')
-    assert.equal(typeis(req, ['text/*']), 'text/html')
+  it('should ignore params', function (done) {
+    createRequest('text/html; charset=utf-8', function (req) {
+      assert.equal(typeis(req, ['text/*']), 'text/html')
+      done()
+    })
   })
 
-  it('should ignore params LWS', function () {
-    var req = createRequest('text/html ; charset=utf-8')
-    assert.equal(typeis(req, ['text/*']), 'text/html')
+  it('should ignore params LWS', function (done) {
+    createRequest('text/html ; charset=utf-8', function (req) {
+      assert.equal(typeis(req, ['text/*']), 'text/html')
+      done()
+    })
   })
 
-  it('should ignore casing', function () {
-    var req = createRequest('text/HTML')
-    assert.equal(typeis(req, ['text/*']), 'text/html')
+  it('should ignore casing', function (done) {
+    createRequest('text/HTML', function (req) {
+      assert.equal(typeis(req, ['text/*']), 'text/html')
+      done()
+    })
   })
 
-  it('should fail invalid type', function () {
-    var req = createRequest('text/html**')
-    assert.strictEqual(typeis(req, ['text/*']), false)
+  it('should fail invalid type', function (done) {
+    createRequest('text/html**', function (req) {
+      assert.strictEqual(typeis(req, ['text/*']), false)
+      done()
+    })
   })
 
-  it('should not match invalid type', function () {
-    var req = createRequest('text/html')
-    assert.strictEqual(typeis(req, ['text/html/']), false)
-    assert.strictEqual(typeis(req, [undefined, null, true, function () {}]), false)
+  it('should not match invalid type', function (done) {
+    createRequest('text/html', function (req) {
+      assert.strictEqual(typeis(req, ['text/html/']), false)
+      assert.strictEqual(typeis(req, [undefined, null, true, function () {}]), false)
+      done()
+    })
   })
 
   describe('when no body is given', function () {
-    it('should return null', function () {
-      var req = {headers: {}}
-
-      assert.strictEqual(typeis(req), null)
-      assert.strictEqual(typeis(req, ['image/*']), null)
-      assert.strictEqual(typeis(req, 'image/*', 'text/*'), null)
+    it('should return null', function (done) {
+      createBodylessRequest('', function (req) {
+        assert.strictEqual(typeis(req), null)
+        assert.strictEqual(typeis(req, ['image/*']), null)
+        assert.strictEqual(typeis(req, 'image/*', 'text/*'), null)
+        done()
+      })
     })
   })
 
   describe('when no content type is given', function () {
-    it('should return false', function () {
-      var req = createRequest()
-      assert.strictEqual(typeis(req), false)
-      assert.strictEqual(typeis(req, ['image/*']), false)
-      assert.strictEqual(typeis(req, ['text/*', 'image/*']), false)
-    })
+    if (!process.env.HTTP2_TEST) {
```

Comment:
Why is this excludes from HTTP/2 tests? To me this seems like it should work in http/2.

Reply:
I agree with you. I just cannot found how to unset content-type in superagent. I'm checking superagent code.